quinn-proto/src/connection/assembler.rs

```diff
@@ -465,6 +465,24 @@ mod test {
         assert_eq!(x.pop(), Some((3, Bytes::from_static(b"def"))));
     }
 
+    #[test]
+    fn defrag_read_chunk() {
+        let mut x = Assembler::new();
+        x.insert(3, Bytes::from_static(b"def"));
+        x.insert(0, Bytes::from_static(b"abc"));
+        x.insert(7, Bytes::from_static(b"ijk"));
+        x.insert(11, Bytes::from_static(b"mno"));
+        x.defragment();
+        assert_matches!(x.read_chunk(), Ok(Some(ref y)) if &y[..] == b"abcdef");
+        x.insert(5, Bytes::from_static(b"fghijklmno"));
```

Comment:
Is this supplying an inconsistent byte at offset 7? Looks like line 473 puts `i` there, and now we're saying it's `h`.

Reply:
You are right. Now I had to do all the complicated spelling again